packages/mobile/src/screens/edit-track-screen/EditTrackFormPreviewContext.tsx

```diff
@@ -27,59 +20,32 @@ export const EditTrackFormPreviewContext = createContext<PreviewContextProps>({
 export const EditTrackFormPreviewContextProvider = (props: {
   children: JSX.Element
 }) => {
-  // const dispatch = useDispatch()
+  const dispatch = useDispatch()
   const [isPlaying, setIsPlaying] = useState(false)
-  // const uid = useSelector(playerSelectors.getUid)
-  // const queueOrder = useSelector(queueSelectors.getOrder)
-  // const queueLength = useSelector(queueSelectors.getLength)
-  // console.log({ uid, queueLength, queueOrder })
+  const isPlayerPlaying = useSelector(playerSelectors.getPlaying)
 
-  // const soundRef = useRef<Sound | null>(null)
+  useEffectOnce(() => {
+    SoundPlayer.setVolume(0.75)
+    SoundPlayer.addEventListener('FinishedPlaying', () => {
+      setIsPlaying(false)
+    })
+  })
 
   // Request preview playback
-  const playPreview = useCallback(async (url: string) => {
-    // console.log('PLAY PREVIEW', { url })
-    setIsPlaying(true)
-
-    // --- 1 ---
-    // const sound = new Sound(url, Sound.MAIN_BUNDLE, (error) => {
-    // const sound = new Sound(url, '', (error) => {
-    //   if (error) {
-    //     console.log('failed to load the sound', error)
-    //     return
-    //   }
-    //   // when loaded successfully
-    //   console.log(
-    //     'preview duration: ' +
-    //       sound.getDuration()
-    //   )
-    // })
-    // sound.setVolume(1)
-    // soundRef.current = sound
-    // soundRef.current.play()
+  const playPreview = useCallback(
+    async (url: string) => {
+      if (isPlayerPlaying) dispatch(playerActions.pause())
```

Comment:
we can check w/product, but I assume we don't need to worry about resuming the track when we end the preview, right?

Sounds more confusing than anything, imo

Reply:
Yea i thought about this too. wrote it down as a sync question